### DISCUSSION TITLE
Issue in Intercommunication between Cyclone DDS and OpenDDS

### DISCUSSION THREAD

**Discussion by Aman568** [OpenDDS/OpenDDS#4626] [Q&A]
Hello,

We are trying to establish communication between Cyclone DDS(V.0.10.4)(Subscriber) and Open DDS(3.26)(Publisher).

In Discovery is enabled over multicast IP in both Cyclone DDS and Open DDS.
Topic is same in both Cyclone DDS and Open DDS.

We could see RTPS packets transmitted by Open DDS and Cyclone DDS.

Issue : Subscriber(Cyclone DDS) is not able to receive the Data transmitted by Publisher(Open DDS).

We are not sure what could be the possible reason behind this issue. However our observation are as follows :

Observation :

Open DDS is using RTPS 2.3 and Cyclone DDS is using version 2.1
Open DDS is using encapsulation kind D_CDR2_LE(0x0009) but Cyclone DDS uses CDR_LE(0x0001) for the data encapsulation.
Open DDS(Publisher) to Open DDS(Subscriber) is working fine.
Cyclone DDS(Publisher) to Cyclone DDS(Subscriber) is working fine.
What else we need to consider when communicating between Cyclone DDS and Open DDS.

Could you support us to resolve intercommunication Issue between Cyclone DDS and Open DDS.

Thank You.
Best Regards
Aman

**Comment by jwillemsen** [comment]
https://github.com/eclipse-cyclonedds/cyclonedds/issues/1897

**Comment by mitza-oci** [comment]
> We are trying to establish communication between Cyclone DDS(V.0.10.4)(Subscriber) and OpenDDS (3.26)(Publisher).
> 
> In Discovery is enabled over multicast IP in both Cyclone DDS and OpenDDS. Topic is same in both Cyclone DDS and OpenDDS.
> 
> We could see RTPS packets transmitted by OpenDDS and Cyclone DDS.
> 
> Issue : Subscriber(Cyclone DDS) is not able to receive the Data transmitted by Publisher(OpenDDS).
> 
> We are not sure what could be the possible reason behind this issue. However our observation are as follows :
> 
> Observation :
> 
> Open DDS is using RTPS 2.3 and Cyclone DDS is using version 2.1 OpenDDS is using encapsulation kind D_CDR2_LE(0x0009) but Cyclone DDS uses CDR_LE(0x0001) for the data encapsulation. OpenDDS (Publisher) to Open DDS(Subscriber) is working fine. Cyclone DDS(Publisher) to Cyclone DDS(Subscriber) is working fine. What else we need to consider when communicating between Cyclone DDS and OpenDDS.

The "encapsulation" kinds correspond to (X)CDR version 1 and XCDR version 2.  The two participants need to agree on an encoding type/version in order to communicate.
